server/sonar-server/src/main/java/org/sonar/server/platform/web/requestid/RequestUidFilter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.platform.web.requestid;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import org.sonar.server.platform.Platform;
+
+/**
+ * A {@link Filter} that puts and removes the HTTP request UID from the {@link org.slf4j.MDC}.
+ */
+public class RequestUidFilter implements Filter {
+  private final Platform platform;
+
+  public RequestUidFilter() {
+    this(Platform.getInstance());
+  }
+
+  @VisibleForTesting
+  RequestUidFilter(Platform platform) {
+    this.platform = platform;
+  }
+
+  @Override
+  public void init(FilterConfig filterConfig) throws ServletException {
+    // nothing to do
+  }
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+    RequestUidGenerator requestUidGenerator = platform.getContainer().getComponentByType(RequestUidGenerator.class);
+    RequestUidMDCStorageFactory requestUidMDCStorageFactory = platform.getContainer().getComponentByType(RequestUidMDCStorageFactory.class);
+
+    try (RequestUidMDCStorage mdcStorage = requestUidMDCStorageFactory.put(requestUidGenerator.generate())) {
```

Comment:
I went for a solution half way: I kept RequestUidMDCStorage to isolate code and responsibility but dropped the factory